core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnBatchCreator.scala

```diff
@@ -106,7 +104,7 @@ final class ColumnBatchCreator(
         // the index of the batchId (and bucketId after that) has already
         // been pushed in during compilation above
         val batchIdRef = references(references.length - 1).asInstanceOf[Int]
-        references(batchIdRef) = Some(batchID.toString)
```

Comment:
Some was not required here in the previous code also. Right?

Reply:
Java generated code used null internally to indicate absence while here it was converted to Some/None. Removing the conversions and instead using invalidUUID (-1) to indicate absence everywhere since that is what the JDBC row during smart connector insert can hold.